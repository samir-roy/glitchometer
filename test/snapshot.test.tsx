import React from 'react';
import renderer from 'react-test-renderer';
import Glitchometer from '../src';

describe('snapshot tests', () => {
  it('should render all digits', () => {
    const tree = renderer.create(<Glitchometer value="91,345.02%" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
