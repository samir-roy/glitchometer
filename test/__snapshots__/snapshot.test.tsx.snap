// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests should render all digits 1`] = `
<div
  style={
    {
      "height": 56,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      {
        "WebkitTransform": "translate(calc(-50% - 3px), 0px)",
        "left": "50%",
        "position": "absolute",
        "transform": "translate(calc(-50% - 3px), 0px)",
        "width": 250,
        "zIndex": 1,
      }
    }
  >
    <div>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 32 54"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          9
        </title>
        <path
          d="M4.74675 1.224C2.37875 1.224 1.80275 1.864 1.80275 4.168V22.536C1.80275 24.904 2.44275 25.48 4.74675 25.48H18.3148C20.4268 25.48 20.8748 25.928 20.8748 28.04V42.056C20.8748 44.36 21.6428 45 23.8188 45H27.2748C29.4508 45 30.2188 44.36 30.2188 42.056V4.168C30.2188 1.8 29.5788 1.224 27.2748 1.224H4.74675ZM22.4108 6.472C24.5228 6.472 24.9708 6.92 24.9708 9.032V17.672C24.9708 19.784 24.5228 20.232 22.4108 20.232H9.61075C7.49875 20.232 7.05075 19.784 7.05075 17.672V9.032C7.05075 6.792 7.56275 6.536 9.35475 6.472H22.4108Z"
          fill="var(--primary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 29 54"
        width="29"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          0
        </title>
        <path
          d="M4.39094 1L24.672 1C26.3708 1 27 1.64327 27 3.38012L27 42.6199C27 44.3567 26.3708 45 24.672 45H4.32802C2.62919 45 2 44.3567 2 42.6199L2 3.44444C2 1.77193 2.75503 1 4.39094 1ZM20.4564 7.16114L8.35486 7.16114C6.8448 7.22547 6.27852 7.54711 6.27852 9.47693L6.27852 38.3099C6.27852 40.1754 6.71896 40.6257 8.54362 40.6257H11.1792C13.0039 40.6257 13.4444 40.1754 13.4444 38.3099L13.4444 27.4529C13.4444 25.3944 14.6398 25.3944 16.087 25.3944C16.1908 25.3944 20.6612 25.3948 20.7592 25.3944C21.9678 25.3899 22.7215 25.2754 22.7215 23.0143L22.7215 9.47693C22.7215 7.61144 22.281 7.16114 20.4564 7.16114Z"
          fill="var(--primary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 14 54"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          ,
        </title>
        <path
          d="M4.42975 32.648C2.70175 32.648 2.31775 33.032 2.31775 34.76V42.888C2.31775 45 2.89375 45 4.94175 45C5.64575 45 6.34975 44.936 6.34975 45.704C6.34975 46.728 5.70975 48.264 3.59775 49.864C2.89375 50.376 2.31775 50.824 2.31775 51.72C2.31775 52.552 2.82975 53.384 4.04575 53.384C8.65375 53.384 11.9178 49.736 11.9178 44.104V34.76C11.9178 33.032 11.5338 32.648 9.80575 32.648H4.42975Z"
          fill="var(--primary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 28 54"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          3
        </title>
        <path
          d="M4.85875 1.224C2.81075 1.224 1.85075 1.992 1.85075 3.848C1.85075 5.704 2.81075 6.472 4.85875 6.472H13.7548C15.8028 6.472 16.3788 7.048 16.3788 9.096V17.864C16.3788 19.912 15.8028 20.488 13.7548 20.488H5.11475C3.06675 20.488 2.10675 21.256 2.10675 23.112C2.10675 24.968 3.06675 25.736 5.11475 25.736H13.7548C15.8028 25.736 16.3788 26.312 16.3788 28.36V37.128C16.3788 39.176 15.8028 39.752 13.7548 39.752H4.85875C2.81075 39.752 1.85075 40.52 1.85075 42.376C1.85075 44.232 2.81075 45 4.85875 45H22.7788C25.0188 45 25.7228 44.296 25.7228 41.992V25.992C25.7228 24.264 25.5308 23.624 23.5468 23.304C21.9468 23.048 21.6268 22.728 21.6268 20.552V4.168C21.6268 1.928 20.9228 1.224 18.6828 1.224H4.85875Z"
          fill="var(--primary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 32 54"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          4
        </title>
        <path
          d="M4.93875 1.224C2.89075 1.224 2.31475 1.8 2.31475 3.848V32.392C2.31475 34.76 2.95475 35.336 5.25875 35.336H18.1868C20.2348 35.336 20.8108 35.912 20.8108 37.96V42.376C20.8108 44.488 21.4508 45 23.4348 45H27.5308C29.5148 45 30.1548 44.488 30.1548 42.376V27.464C30.1548 25.416 29.5788 24.84 27.5308 24.84H23.4988C20.9388 24.84 20.8108 25.736 20.8108 27.656C20.8108 29.128 20.8108 30.088 18.1228 30.088H14.2828C12.2348 30.088 11.6588 29.512 11.6588 27.464V3.784C11.6588 1.608 11.1468 1.224 9.09875 1.224H4.93875Z"
          fill="var(--primary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 28 54"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          5
        </title>
        <path
          d="M5.24275 1.224C3.06675 1.224 2.23475 2.056 2.23475 4.232V22.728C2.23475 24.904 3.06675 25.736 5.24275 25.736H13.8828C15.9308 25.736 16.5068 26.312 16.5068 28.36V37.128C16.5068 39.176 15.9308 39.752 13.8828 39.752H4.28275C2.29875 39.752 1.40275 40.136 1.40275 41.544V43.208C1.40275 44.616 2.29875 45 4.28275 45H22.8428C25.0188 45 25.8508 44.168 25.8508 41.992V23.496C25.8508 21.32 25.0188 20.488 22.8428 20.488H10.1068C8.05875 20.488 7.48275 19.912 7.48275 17.864V9.096C7.48275 7.048 8.05875 6.472 10.1068 6.472H22.5868C24.5708 6.472 25.4668 6.088 25.4668 4.68V3.016C25.4668 1.608 24.5708 1.224 22.5868 1.224H5.24275Z"
          fill="var(--primary-glitch-color)"
        />
      </svg>
    </div>
  </div>
  <div
    style={
      {
        "WebkitTransform": "translate(calc(-50% + 3px), 0px)",
        "left": "50%",
        "position": "absolute",
        "transform": "translate(calc(-50% + 3px), 0px)",
        "width": 250,
        "zIndex": 1,
      }
    }
  >
    <div>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 32 54"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          9
        </title>
        <path
          d="M4.74675 1.224C2.37875 1.224 1.80275 1.864 1.80275 4.168V22.536C1.80275 24.904 2.44275 25.48 4.74675 25.48H18.3148C20.4268 25.48 20.8748 25.928 20.8748 28.04V42.056C20.8748 44.36 21.6428 45 23.8188 45H27.2748C29.4508 45 30.2188 44.36 30.2188 42.056V4.168C30.2188 1.8 29.5788 1.224 27.2748 1.224H4.74675ZM22.4108 6.472C24.5228 6.472 24.9708 6.92 24.9708 9.032V17.672C24.9708 19.784 24.5228 20.232 22.4108 20.232H9.61075C7.49875 20.232 7.05075 19.784 7.05075 17.672V9.032C7.05075 6.792 7.56275 6.536 9.35475 6.472H22.4108Z"
          fill="var(--secondary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 29 54"
        width="29"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          0
        </title>
        <path
          d="M4.39094 1L24.672 1C26.3708 1 27 1.64327 27 3.38012L27 42.6199C27 44.3567 26.3708 45 24.672 45H4.32802C2.62919 45 2 44.3567 2 42.6199L2 3.44444C2 1.77193 2.75503 1 4.39094 1ZM20.4564 7.16114L8.35486 7.16114C6.8448 7.22547 6.27852 7.54711 6.27852 9.47693L6.27852 38.3099C6.27852 40.1754 6.71896 40.6257 8.54362 40.6257H11.1792C13.0039 40.6257 13.4444 40.1754 13.4444 38.3099L13.4444 27.4529C13.4444 25.3944 14.6398 25.3944 16.087 25.3944C16.1908 25.3944 20.6612 25.3948 20.7592 25.3944C21.9678 25.3899 22.7215 25.2754 22.7215 23.0143L22.7215 9.47693C22.7215 7.61144 22.281 7.16114 20.4564 7.16114Z"
          fill="var(--secondary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 14 54"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          ,
        </title>
        <path
          d="M4.42975 32.648C2.70175 32.648 2.31775 33.032 2.31775 34.76V42.888C2.31775 45 2.89375 45 4.94175 45C5.64575 45 6.34975 44.936 6.34975 45.704C6.34975 46.728 5.70975 48.264 3.59775 49.864C2.89375 50.376 2.31775 50.824 2.31775 51.72C2.31775 52.552 2.82975 53.384 4.04575 53.384C8.65375 53.384 11.9178 49.736 11.9178 44.104V34.76C11.9178 33.032 11.5338 32.648 9.80575 32.648H4.42975Z"
          fill="var(--secondary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 28 54"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          3
        </title>
        <path
          d="M4.85875 1.224C2.81075 1.224 1.85075 1.992 1.85075 3.848C1.85075 5.704 2.81075 6.472 4.85875 6.472H13.7548C15.8028 6.472 16.3788 7.048 16.3788 9.096V17.864C16.3788 19.912 15.8028 20.488 13.7548 20.488H5.11475C3.06675 20.488 2.10675 21.256 2.10675 23.112C2.10675 24.968 3.06675 25.736 5.11475 25.736H13.7548C15.8028 25.736 16.3788 26.312 16.3788 28.36V37.128C16.3788 39.176 15.8028 39.752 13.7548 39.752H4.85875C2.81075 39.752 1.85075 40.52 1.85075 42.376C1.85075 44.232 2.81075 45 4.85875 45H22.7788C25.0188 45 25.7228 44.296 25.7228 41.992V25.992C25.7228 24.264 25.5308 23.624 23.5468 23.304C21.9468 23.048 21.6268 22.728 21.6268 20.552V4.168C21.6268 1.928 20.9228 1.224 18.6828 1.224H4.85875Z"
          fill="var(--secondary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 32 54"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          4
        </title>
        <path
          d="M4.93875 1.224C2.89075 1.224 2.31475 1.8 2.31475 3.848V32.392C2.31475 34.76 2.95475 35.336 5.25875 35.336H18.1868C20.2348 35.336 20.8108 35.912 20.8108 37.96V42.376C20.8108 44.488 21.4508 45 23.4348 45H27.5308C29.5148 45 30.1548 44.488 30.1548 42.376V27.464C30.1548 25.416 29.5788 24.84 27.5308 24.84H23.4988C20.9388 24.84 20.8108 25.736 20.8108 27.656C20.8108 29.128 20.8108 30.088 18.1228 30.088H14.2828C12.2348 30.088 11.6588 29.512 11.6588 27.464V3.784C11.6588 1.608 11.1468 1.224 9.09875 1.224H4.93875Z"
          fill="var(--secondary-glitch-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 28 54"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          5
        </title>
        <path
          d="M5.24275 1.224C3.06675 1.224 2.23475 2.056 2.23475 4.232V22.728C2.23475 24.904 3.06675 25.736 5.24275 25.736H13.8828C15.9308 25.736 16.5068 26.312 16.5068 28.36V37.128C16.5068 39.176 15.9308 39.752 13.8828 39.752H4.28275C2.29875 39.752 1.40275 40.136 1.40275 41.544V43.208C1.40275 44.616 2.29875 45 4.28275 45H22.8428C25.0188 45 25.8508 44.168 25.8508 41.992V23.496C25.8508 21.32 25.0188 20.488 22.8428 20.488H10.1068C8.05875 20.488 7.48275 19.912 7.48275 17.864V9.096C7.48275 7.048 8.05875 6.472 10.1068 6.472H22.5868C24.5708 6.472 25.4668 6.088 25.4668 4.68V3.016C25.4668 1.608 24.5708 1.224 22.5868 1.224H5.24275Z"
          fill="var(--secondary-glitch-color)"
        />
      </svg>
    </div>
  </div>
  <div
    style={
      {
        "WebkitTransform": "translate(calc(-50% + 0px), 0px)",
        "left": "50%",
        "position": "absolute",
        "transform": "translate(calc(-50% + 0px), 0px)",
        "width": 250,
        "zIndex": 2,
      }
    }
  >
    <div>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 32 54"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          9
        </title>
        <path
          d="M4.74675 1.224C2.37875 1.224 1.80275 1.864 1.80275 4.168V22.536C1.80275 24.904 2.44275 25.48 4.74675 25.48H18.3148C20.4268 25.48 20.8748 25.928 20.8748 28.04V42.056C20.8748 44.36 21.6428 45 23.8188 45H27.2748C29.4508 45 30.2188 44.36 30.2188 42.056V4.168C30.2188 1.8 29.5788 1.224 27.2748 1.224H4.74675ZM22.4108 6.472C24.5228 6.472 24.9708 6.92 24.9708 9.032V17.672C24.9708 19.784 24.5228 20.232 22.4108 20.232H9.61075C7.49875 20.232 7.05075 19.784 7.05075 17.672V9.032C7.05075 6.792 7.56275 6.536 9.35475 6.472H22.4108Z"
          fill="var(--text-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 29 54"
        width="29"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          0
        </title>
        <path
          d="M4.39094 1L24.672 1C26.3708 1 27 1.64327 27 3.38012L27 42.6199C27 44.3567 26.3708 45 24.672 45H4.32802C2.62919 45 2 44.3567 2 42.6199L2 3.44444C2 1.77193 2.75503 1 4.39094 1ZM20.4564 7.16114L8.35486 7.16114C6.8448 7.22547 6.27852 7.54711 6.27852 9.47693L6.27852 38.3099C6.27852 40.1754 6.71896 40.6257 8.54362 40.6257H11.1792C13.0039 40.6257 13.4444 40.1754 13.4444 38.3099L13.4444 27.4529C13.4444 25.3944 14.6398 25.3944 16.087 25.3944C16.1908 25.3944 20.6612 25.3948 20.7592 25.3944C21.9678 25.3899 22.7215 25.2754 22.7215 23.0143L22.7215 9.47693C22.7215 7.61144 22.281 7.16114 20.4564 7.16114Z"
          fill="var(--text-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 14 54"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          ,
        </title>
        <path
          d="M4.42975 32.648C2.70175 32.648 2.31775 33.032 2.31775 34.76V42.888C2.31775 45 2.89375 45 4.94175 45C5.64575 45 6.34975 44.936 6.34975 45.704C6.34975 46.728 5.70975 48.264 3.59775 49.864C2.89375 50.376 2.31775 50.824 2.31775 51.72C2.31775 52.552 2.82975 53.384 4.04575 53.384C8.65375 53.384 11.9178 49.736 11.9178 44.104V34.76C11.9178 33.032 11.5338 32.648 9.80575 32.648H4.42975Z"
          fill="var(--text-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 28 54"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          3
        </title>
        <path
          d="M4.85875 1.224C2.81075 1.224 1.85075 1.992 1.85075 3.848C1.85075 5.704 2.81075 6.472 4.85875 6.472H13.7548C15.8028 6.472 16.3788 7.048 16.3788 9.096V17.864C16.3788 19.912 15.8028 20.488 13.7548 20.488H5.11475C3.06675 20.488 2.10675 21.256 2.10675 23.112C2.10675 24.968 3.06675 25.736 5.11475 25.736H13.7548C15.8028 25.736 16.3788 26.312 16.3788 28.36V37.128C16.3788 39.176 15.8028 39.752 13.7548 39.752H4.85875C2.81075 39.752 1.85075 40.52 1.85075 42.376C1.85075 44.232 2.81075 45 4.85875 45H22.7788C25.0188 45 25.7228 44.296 25.7228 41.992V25.992C25.7228 24.264 25.5308 23.624 23.5468 23.304C21.9468 23.048 21.6268 22.728 21.6268 20.552V4.168C21.6268 1.928 20.9228 1.224 18.6828 1.224H4.85875Z"
          fill="var(--text-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 32 54"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          4
        </title>
        <path
          d="M4.93875 1.224C2.89075 1.224 2.31475 1.8 2.31475 3.848V32.392C2.31475 34.76 2.95475 35.336 5.25875 35.336H18.1868C20.2348 35.336 20.8108 35.912 20.8108 37.96V42.376C20.8108 44.488 21.4508 45 23.4348 45H27.5308C29.5148 45 30.1548 44.488 30.1548 42.376V27.464C30.1548 25.416 29.5788 24.84 27.5308 24.84H23.4988C20.9388 24.84 20.8108 25.736 20.8108 27.656C20.8108 29.128 20.8108 30.088 18.1228 30.088H14.2828C12.2348 30.088 11.6588 29.512 11.6588 27.464V3.784C11.6588 1.608 11.1468 1.224 9.09875 1.224H4.93875Z"
          fill="var(--text-color)"
        />
      </svg>
      <svg
        fill="none"
        height="54"
        viewBox="0 0 8 54"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        fill="none"
        height="54"
        viewBox="0 0 28 54"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          5
        </title>
        <path
          d="M5.24275 1.224C3.06675 1.224 2.23475 2.056 2.23475 4.232V22.728C2.23475 24.904 3.06675 25.736 5.24275 25.736H13.8828C15.9308 25.736 16.5068 26.312 16.5068 28.36V37.128C16.5068 39.176 15.9308 39.752 13.8828 39.752H4.28275C2.29875 39.752 1.40275 40.136 1.40275 41.544V43.208C1.40275 44.616 2.29875 45 4.28275 45H22.8428C25.0188 45 25.8508 44.168 25.8508 41.992V23.496C25.8508 21.32 25.0188 20.488 22.8428 20.488H10.1068C8.05875 20.488 7.48275 19.912 7.48275 17.864V9.096C7.48275 7.048 8.05875 6.472 10.1068 6.472H22.5868C24.5708 6.472 25.4668 6.088 25.4668 4.68V3.016C25.4668 1.608 24.5708 1.224 22.5868 1.224H5.24275Z"
          fill="var(--text-color)"
        />
      </svg>
    </div>
  </div>
</div>
`;
